### PR TITLE
Add mapping for CAPI -> AzureConfig + AzureClusterConfig CR

### PR DESCRIPTION
**Pull request by tuommaki**
In order to reduce control plane related changes when implementing Node Pools
on top of CAPI & CAPZ CRDs, we maintain mapping to AzureClusterConfig CR so
that cluster-operator is still using the old code for cert management.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -157,7 +157,6 @@ func mainError() error {
 	daemonCommand.PersistentFlags().Int(f.Service.Cluster.Kubernetes.Kubelet.Port, 0, "Port to bind guest cluster kubelets on.")
 	daemonCommand.PersistentFlags().String(f.Service.Cluster.Kubernetes.NetworkSetup.Docker.Image, "", "Full docker image of networksetup.")
 	daemonCommand.PersistentFlags().String(f.Service.Cluster.Kubernetes.SSH.UserList, "", "Comma separated list of ssh users and their public key in format `username:publickey`, being installed in the guest cluster nodes.")
-	daemonCommand.PersistentFlags().String(f.Service.Installation.Guest.IPAM.Network.CIDR, "", "Guest cluster network segment from which IPAM allocates subnets.")
```
Comment (tuommaki) [inline main.go:160]: I probably fat fingered some merge conflict at some point...

File: service/controller/resource/azureclusterconfig/create.go [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling reconciliation")
+			reconciliationcanceledcontext.SetCanceled(ctx)
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var masterMachines []capzv1alpha3.AzureMachine
+	var workerMachines []capzv1alpha3.AzureMachine
```
Comment (tuommaki) [inline service/controller/resource/azureclusterconfig/create.go:51]: Currently this expected `AzureMachine` CRs for worker nodes that exists in the built-in worker VMSS (i.e. which is not node pool).

**Comment by tuommaki** [comment]
This is one part of the mapping. Contains also few minor merge conflict fixes. Another part is the destination branch of this.

File: service/controller/resource/azureclusterconfig/availability_zones_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+			input:    []int{1, 4, 2, 3},
+			expected: []int{1, 2, 3, 4},
+		},
+		{
+			name:     "case 5: multi element sorted non-uniq slice",
+			input:    []int{1, 2, 2, 2, 3, 3, 4},
+			expected: []int{1, 2, 3, 4},
+		},
+		{
+			name:     "case 4: multi element unsorted non-uniq slice",
```
Comment (axbarsan) [inline service/controller/resource/azureclusterconfig/availability_zones_test.go:47]:
```suggestion
			name:     "case 6: multi element unsorted non-uniq slice",
```

Comment (tuommaki) [reply service/controller/resource/azureclusterconfig/availability_zones_test.go:47]: :see_no_evil: Thanks!

File: service/controller/resource/azureclusterconfig/create.go [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+
+		if len(masterMachines) < 1 {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no control plane AzureMachines found for cluster %q", key.ClusterID(&cluster)))
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cancelling reconciliation")
+			reconciliationcanceledcontext.SetCanceled(ctx)
+			return nil
+		}
+
+		if len(workerMachines) < 1 {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("worker AzureMachines found for cluster %q", key.ClusterID(&cluster)))
```
Comment (fiunchinho) [inline service/controller/resource/azureclusterconfig/create.go:82]:
```suggestion
			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no worker AzureMachines found for cluster %q", key.ClusterID(&cluster)))
```

File: service/controller/resource/azureclusterconfig/create.go [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+		return domainLabels[0]
+	case len(domainLabels[0]) == 0 || len(domainLabels[1]) == 0:
+		return apiEndpointHost
+	default:
+		// Drop first label of domain (e.g. api.foobar.com -> foobar.com).
+		return strings.Join(domainLabels[1:], ".")
+	}
+}
+
+func findComponentVersion(component string, releaseComponents []releasev1alpha1.ReleaseSpecComponent) (string, error) {
```
Comment (fiunchinho) [inline service/controller/resource/azureclusterconfig/create.go:253]: I think we have a function for this in the `key` package?

File: service/controller/azure_cluster_resource_set.go [hunk 6]
```diff
@@ -51,37 +54,24 @@ func NewAzureClusterResourceSet(config AzureClusterResourceSetConfig) (*controll
...
-				Logger:    config.Logger,
+	var azureClusterConfigResource *azureclusterconfig.Resource
+	{
+		c := azureclusterconfig.Config{
+			Logger: config.Logger,
 
-				WatchTimeout: 5 * time.Second,
-			}
+			Flag:  config.Flag,
+			Viper: config.Viper,
```
Comment (fiunchinho) [inline service/controller/azure_cluster_resource_set.go:63]: Same I mentioned in the other PR.

File: service/controller/resource/azureclusterconfig/create.go (hunk 5 above)
Comment (tuommaki) [reply service/controller/resource/azureclusterconfig/create.go:253]: I couldn't find. Did you possibly see it in `k8scloudconfig`: https://github.com/giantswarm/k8scloudconfig/blob/master/pkg/template/release.go#L66 ?

Comment (fiunchinho) [reply service/controller/resource/azureclusterconfig/create.go:253]: I was thinking about https://github.com/giantswarm/azure-operator/blob/4471d37c30d2570f7ecf02d9701e0b37e0549cc2/service/controller/key/key.go#L191-L198

Comment (tuommaki) [reply service/controller/resource/azureclusterconfig/create.go:253]: Nice catch! Got it.

**Comment by tuommaki** [comment]
Works with CAPI CRs & traditional `api` created `AzureConfigs`.

**Pull request merged**